jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpURI.java

```diff
@@ -151,6 +151,11 @@ static Immutable from(String scheme, String host, int port, String pathQuery)
         return new Mutable(scheme, host, port, pathQuery).asImmutable();
     }
 
+    static Immutable from(String scheme, String host, int port, String path, String query, String fragment)
+    {
+        return new Mutable(scheme, host, port, path, query, fragment).asImmutable();
```

Comment:
Couldn't this be ...

```suggestion
        return new Immutable(scheme, host, port, path, query, fragment);
```